src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/Infrastructure/TaskExtensions.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.ServiceBus.UnitTests
+{
+    /// <summary>
+    ///   The set of extension methods for the <see cref="Task" /> class.
+    /// </summary>
+    /// 
+    public static class TaskExtensions
+    {
+        /// <summary>
+        ///   Executes a task without waiting for or observing its completion.
+        /// </summary>
+        /// 
+        /// <param name="instance">The target of this fire-and-forget.</param>
+        /// <param name="errorAction">An action to execute should an exception occur.</param>
+        /// 
+        public static async void FireAndForget(this Task instance, Action<Exception> errorAction = null)
+        {
+            if (instance == null)
+            {
+                return;
+            }
+
+            try
+            {
+                await instance.ConfigureAwait(false);
+            }
+
+            catch (Exception ex)
+            {
+                errorAction?.Invoke(ex);
+            }
+        }
+
+        /// <summary>
+        ///   Executes a task without waiting for or observing its completion.
+        /// </summary>
+        /// 
+        /// <typeparam name="T">The type of return value of the task.</typeparam>
+        /// 
+        /// <param name="instance">The target of this fire-and-forget.</param>
+        /// <param name="errorAction">An action to execute should an exception occur.</param>
+        /// 
+        public static async void FireAndForget<T>(this Task<T> instance, Action<Exception> errorAction = null)
```

Comment:
I strongly suggest that we should avoid `async void` methods, you might know that they are not allowed to throw but someone might copy them and modify in a way that would allow throwing and cause very hard to diagnose test crashes. `_ = obj.DoSomethingAsync();` accomplishes the same thing.

Reply:
In a normal context, I'd agree.   This extension is explicitly intended to allow the task to complete unobserved while providing a means to discover and act on any exceptions that arise in its execution.   

If I understand your suggestion,  The missing component there is that the task for `DoSomethingAsync` will continue unobserved but lacks the ability to report on exceptions.  Given that it's not being returned, I'm not sure I follow why that would be a better pattern.   Perhaps I'm missing something, could you expand on what you're thinking here, @pakrym?

Reply:
`async void` is not something worth using even in a strong team, there are no benefits to it, it just produces patterns that have to be fixed later. FireAndForget with error callback is one of them.

> If I understand your suggestion correctly, the missing component that I see is that, in the proposed approach, the task for DoSomethingAsync will continue unobserved but lacks the ability to report on exceptions. Given that it's not being returned, I'm not sure I follow why that would be a better pattern. Perhaps I'm missing something, could you expand on what you're thinking here, @pakrym?

If we want to observe the exception we shouldn't fire and forget but rather just await the task as part of the test. If we want don't care about the exception assigning task to _ would do it. Do we use error reporting callback anywhere in this change?

Reply:
We'll have to agree to disagree on that point.   That said, looking over the usage, it's only in one spot outside this class, so I'll concede that it's easy enough to work around in this case, albeit with less clear semantics, and we can save the larger philosophical discussion for another time.    :)